mobile/src/main/java/org/openhab/habdroid/background/AlarmChangedWorker.java

```diff
@@ -0,0 +1,134 @@
+
+import android.app.AlarmManager;
+import android.app.Notification;
+import android.app.NotificationManager;
+import android.content.Context;
+import android.os.Build;
+import android.preference.PreferenceManager;
+import android.text.TextUtils;
+import android.util.Log;
+import androidx.annotation.CheckResult;
+import androidx.annotation.NonNull;
+import androidx.annotation.RequiresApi;
+import androidx.core.app.NotificationCompat;
+import androidx.work.Result;
+import androidx.work.Worker;
+import androidx.work.WorkerParameters;
+
+import org.openhab.habdroid.R;
+import org.openhab.habdroid.core.connection.Connection;
+import org.openhab.habdroid.core.connection.ConnectionFactory;
+import org.openhab.habdroid.core.connection.exception.ConnectionException;
+import org.openhab.habdroid.util.SyncHttpClient;
+
+import java.util.Locale;
+
+import static android.content.Context.NOTIFICATION_SERVICE;
+import static org.openhab.habdroid.background.BackgroundUtils.NOTIFICATION_ID_SEND_ALARM_CLOCK;
+import static org.openhab.habdroid.background.BackgroundUtils.NOTIFICATION_TAG_BACKGROUND;
+import static org.openhab.habdroid.background.BackgroundUtils.NOTIFICATION_TAG_BACKGROUND_ERROR;
+import static org.openhab.habdroid.util.Constants.PREFERENCE_ALARM_CLOCK_ITEM;
+
+public class AlarmChangedWorker extends Worker {
+    private static final String TAG = AlarmChangedWorker.class.getSimpleName();
+    private static final int MAX_RETRY = 3;
+
+    public AlarmChangedWorker(
+            @NonNull Context context,
+            @NonNull WorkerParameters params) {
+        super(context, params);
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
+    @Override
+    public Result doWork() {
+        Context context = getApplicationContext();
+        NotificationManager nm =
+                (NotificationManager) context.getSystemService(NOTIFICATION_SERVICE);
+
+        Connection connection;
+        ConnectionException failureReason;
+        try {
+            Log.d(TAG, "Trying to get connection");
+            connection = ConnectionFactory.getUsableConnection();
+            failureReason = null;
+        } catch (ConnectionException e) {
+            Log.e(TAG, "ConnectionException", e);
+            connection = null;
+            failureReason = e;
+        }
+
+        if (connection == null) {
+            Log.e(TAG, "Got no connection "
+                    + (failureReason != null ? failureReason.getMessage() : null));
+            return retryOrFail(
+                    context.getString(R.string.error_sending_alarm_clock_no_connection),
+                    context, nm);
+        }
+
+        AlarmManager alarmManager = (AlarmManager) context.getSystemService(Context.ALARM_SERVICE);
+        String nextAlarm;
+        if (alarmManager.getNextAlarmClock() == null) {
+            nextAlarm = "0";
+        } else {
+            nextAlarm = String.valueOf(alarmManager.getNextAlarmClock().getTriggerTime());
```

Comment:
Is there already some predecent for this format? IOW, will consumers of this item understand it? Does this need to be documented?

Reply:
This needs documentation and a sample rule to parse the time.